doc/source/batch-inference/preprocessing.rst

```diff
@@ -0,0 +1,118 @@
+.. _inference_preprocessing:
+
+Preprocessing Data
+==================
+
+After loading in a Dataset, the input the input data often needs to be preprocessed prior to inference. This may include cropping or resizing images, or tokenizing raw text.
+
+With :ref:`Ray Data <datasets>`, you can define user-defined functions (UDFs) that transform batches of your data. Applying these UDFs via :meth:`ds.map_batches() <ray.data.Dataset.map_batches>` will output a new, transformed Dataset.
+
+.. tip::
+    
+    For the full suite of Dataset transformations, read :ref:`transforming_datasets`.
+
+
+Writing batch UDFs
+------------------
+Mapping a UDF over batches is the simples transform for Ray Datasets.
+
+The UDF defines the logic for transforming individual batches of data of the dataset Performing operations over batches of data is more performant than single element operations as it can leverage the underlying vectorization capabilities of Pandas or Numpy.
+
+The following is an example to make use of those transformation APIs for processing
+the Iris dataset.
+
+.. literalinclude:: ../data/doc_code/transforming_datasets.py
```

Comment:
Should also give an example of how to pass a custom constructor arg / custom call arg. It comes up often in feedback.